### PR TITLE
New resource `azurerm_api_management_redis_cache`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/4765

--- PASS: TestAccApiManagementCache_basic (1980.64s)
--- PASS: TestAccApiManagementCache_requiresImport (2013.39s)
--- PASS: TestAccApiManagementCache_complete (1991.81s)
--- PASS: TestAccApiManagementCache_update (2277.54s)

### CODE REVIEW

File: website/docs/r/api_management_cache.html.markdown [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+* `connection_string` - (Required) The connection string to the Cache for Redis.
+
+---
+
+* `description` - (Optional) The description of the API Management Cache.
+
+* `redis_cache_id` - (Optional) The resource ID of the Cache for Redis.
+
+* `use_from_location` - (Optional) The location where to use cache from. Possible values are `default` and valid Azure regions. Defaults to `default`.
```
Comment (katbyte) [inline website/docs/r/api_management_cache.html.markdown:70]: this is not the best name, what about
```suggestion
* `cache_location` - (Optional) The location where to use cache from. Possible values are `default` and valid Azure regions. Defaults to `default`.
```

also how is this different from the redis cache location

File: azurerm/internal/services/apimanagement/registration.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
 		"azurerm_api_management_api_version_set":             resourceApiManagementApiVersionSet(),
 		"azurerm_api_management_authorization_server":        resourceApiManagementAuthorizationServer(),
 		"azurerm_api_management_backend":                     resourceApiManagementBackend(),
+		"azurerm_api_management_cache":                       resourceApiManagementCache(),
```
Comment (katbyte) [inline azurerm/internal/services/apimanagement/registration.go:46]: is this always redis?
```suggestion
		"azurerm_api_management_redis_cache":                       resourceApiManagementRedisCache(),
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by yupwei68** [comment]
Thanks kt, Please continue reviewing.

**Comment by katbyte** [comment]
thanks, i do want to know why we have a `cache_location` property - how is this different from the redis location and cannot we not just query the redis cache to get the location from it?

**Comment by yupwei68** [comment]
Hi kt, I think in that way, we could not set a cache with useFromLocation "default". I think the "default" cache is for apim service in non-specified location.

Here is from [MS document](https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-cache-external#-add-an-external-cache)

> The Use from setting specifies an Azure region or a self-hosted gateway location that will use the configured cache. The caches configured as Default will be overridden by caches with a specific matching region or location value.
> 
> For example, if API Management is hosted in the East US, Southeast Asia and West Europe regions and there are two caches configured, one for Default and one for Southeast Asia, API Management in Southeast Asia will use its own cache, while the other two regions will use the Default cache entry.

**Review by katbyte — APPROVED** [review]
LGTM 👍

**Pull request merged**